### PR TITLE
Remove Docker

### PR DESCRIPTION
**Pull request by WiktorOgrodnik**
- removes docker container, the action is now running directly on top Ubuntu VM and linux images are uploaded to the releases
- run-locally script is now base on nektos/act
- the directory tree is flattened
- increase linux boot timeout from 120s to 240s
- changes to host iptables
- move network configuration from shell script to run-in-renode.py
- adds new input: image-path to specify own kernel image
- removes unused toolchain packages from linux image
- only one workflow
- fixes to readme

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -1,16 +1,52 @@
...
-# It is not possible to test a pull request using run_action.yml 
-# because of the action uses the deployed docker image from GHCR.
-
 jobs:
-  default-configuration-test-on-pull-request:
+  build-and-test:
     runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    # if: ${{ !github.event.act }}
```
Comment (kbieganski) [inline .github/workflows/pull_request.yml:9]: Is this a leftover?

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -10,28 +12,51 @@ jobs:
...
+         bc cpio rsync
+
+      - name: install-buildroot
+        run: git clone --branch 2022.11.2 https://github.com/buildroot/buildroot buildroot
+
+      - name: build-kernel
+        run: |
+          cd buildroot && \
+          make BR2_EXTERNAL=../br2-external hifive_unleashed_defconfig && \
+          make -j4 && \
```
Comment (kbieganski) [inline .github/workflows/build_and_test.yml:43]: Why not `-j$(nproc)`?

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -1,16 +1,52 @@
...
+      - name: build-kernel
+        run: |
+          cd buildroot && \
+          make BR2_EXTERNAL=../br2-external hifive_unleashed_defconfig && \
+          make -j4 && \
+          cd .. && \
+          mkdir images && \
+          for i in hifive-unleashed-a00.dtb fw_payload.elf Image rootfs.cpio; do cp buildroot/output/images/$i images; done
+      - name: compress
+        run: tar cJvf images.tar.xz images
```
Comment (kbieganski) [inline .github/workflows/pull_request.yml:30]: If these steps repeat, why not put them in scripts? E.g. some projects have a top-level `ci` dir with scripts. These scripts are then simply called from GHA workflows. We can use the `scripts` directory.

File: .github/workflows/build_and_test.yml [hunk 4]
```diff
@@ -10,28 +12,51 @@ jobs:
...
-      - name: Login to ghcr.io and release
-        uses: pyTooling/Actions/with-post-step@r0
+
+      - name: get packages
+        if: ${{ env.ACT }}
+        run: |
+         sudo apt update -qq && sudo DEBIAN_FRONTEND=noninteractive apt-get -y install --no-install-recommends \
+         bc cpio rsync
+
+      - name: install-buildroot
```
Comment (kbieganski) [inline .github/workflows/build_and_test.yml:36]:
```suggestion
      - name: install buildroot
```

File: .github/workflows/pull_request.yml [hunk 5]
```diff
@@ -1,16 +1,52 @@
...
+
+      - run: git clone https://github.com/antmicro/pyrav4l2.git tests/pyrav4l2
 
-      - name: Bash script
-        run: bash ./run-locally.sh
+      - name: run-in-renode
```
Comment (kbieganski) [inline .github/workflows/pull_request.yml:34]:
```suggestion
      - name: test
```

File: action.yml [hunk 6]
```diff
@@ -10,16 +10,38 @@ inputs:
     description: Devices to set up
     required: false
     default: ""
+  image-source-type:
+    description: How linux image is provided - url or path
+    required: false
+    default: "url"
+  image-source:
```
Comment (kbieganski) [inline action.yml:17]:
```suggestion
  kernel:
```
`image` is pretty generic, and `source` sounds like it wants source code.

File: br2-external/configs/hifive_unleashed_defconfig [hunk 7]
```diff
@@ -64,16 +65,14 @@ BR2_PACKAGE_HOST_UBOOT_TOOLS_BOOT_SCRIPT_SOURCE="board/sifive/hifive-unleashed/b
...
-BR2_TOOLCHAIN_BUILDROOT_CXX=y
-BR2_TOOLCHAIN_BUILDROOT_WCHAR=y
-
 # Packages
 
 # video4linux packages
 BR2_PACKAGE_LIBV4L=y
 BR2_PACKAGE_LIBV4L_UTILS=y
 
+#no
```
Comment (kbieganski) [inline br2-external/configs/hifive_unleashed_defconfig:74]: What's this?

**Review by kbieganski — CHANGES_REQUESTED** [review]
Overall looks good, especially love the option of passing your own kernel.

File: .github/workflows/pull_request.yml (hunk 3 above)
Comment (WiktorOgrodnik) [reply .github/workflows/pull_request.yml:30]: You're right, it would be better

File: action.yml (hunk 6 above)
Comment (WiktorOgrodnik) [reply action.yml:17]: It is not a kernel, but whole linux image with busybox, bootloader, tools, etc.

File: tests/gpio.sh [hunk 8]
```diff
@@ -4,7 +4,7 @@ print_result()
         echo Done!
     else
         echo FAIL!
-        exit 1
+        exit 1face.preferred_buffer_transfo
```
Comment (kbieganski) [inline tests/gpio.sh:7]: What's this?

File: scripts/compile.sh [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+make -j$(nproc)
+
+cd ..
+
+if [ -d images ]; then
+    echo directory images can not exist!
+    exit 1
+fi
+
+mkdir images
```
Comment (kbieganski) [inline scripts/compile.sh:16]:
```suggestion
mkdir -p images
```

File: scripts/get-image.sh [hunk 10]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/bash
+
+IMAGE="$1"
+
+if [ -f "$IMAGE" ]; then
+    if [ $(basename "$IMAGE") == "images.tar.xz" ] && [ -f "./images.tar.xz" ] && [ "$(diff "$IMAGE" ./images.tar.xz)" == "" ]; then
+        echo File already exists
+    else
+        cp "$IMAGE" ./images.tar.xz
+    fi
```
Comment (kbieganski) [inline scripts/get-image.sh:10]: What's the point of this check? Why not just overwrite?

**Review by kbieganski — CHANGES_REQUESTED** [review]

**Review by kbieganski — APPROVED** [review]

**Pull request merged**